src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -213,23 +164,87 @@
 
         <slot name="secondary-actions" />
       </div>
+
+      <!--
+        Instead of using <pre> directly, the <component is="pre"> here acts as a workaround for
+        a potential bug of Vue itself. Because the <Virtualizer> fails to render its scoped slots
+        when it's wrapped inside a <pre> element.
+      -->
+      <!-- eslint-disable-next-line vue/require-component-is -->
+      <component
+        is="pre"
+        v-if="isShowingFilteredCode || hasRenderedFilteredCode"
+        v-show="isShowingFilteredCode"
+        class="filtered-code-block"
+        data-testid="filtered-code-block"
+      >
+        <Virtualizer
+          v-if="!singleLine"
+          v-slot="{ item: line }"
+          v-bind="getVirtualizerProps(true)"
+        >
+          <a
+            :id="`${linePrefix}-L${line}`"
+            class="line-anchor"
+            :href="showLineNumberLinks ? `#${linePrefix}-L${line}` : undefined"
+          >{{ line }}</a>
+        </Virtualizer>
+        <!-- eslint-disable-next-line vue/no-v-html -->
+        <code v-html="filteredCode" />
+      </component>
+
+      <!-- eslint-disable-next-line vue/require-component-is -->
+      <component
+        is="pre"
+        v-if="!isShowingFilteredCode || hasRenderedCode"
+        v-show="!isShowingFilteredCode"
+        class="highlighted-code-block"
+        :class="{
+          'single-line': singleLine
+        }"
+        data-testid="highlighted-code-block"
+      >
+        <Virtualizer
+          v-if="!singleLine"
+          ref="codeBlockLineNumbers"
+          v-slot="{ item: line }"
+          v-bind="getVirtualizerProps(false)"
+        >
+          <a
+            :id="`${linePrefix}-L${line}`"
+            class="line-anchor"
+            :class="{ 'hide-links': !showLineNumberLinks }"
+            :href="showLineNumberLinks ? `#${linePrefix}-L${line}` : undefined"
+          >{{ line }}</a>
```

Comment:
suggestion: should the anchor here have an `aria` attribute(s) to indicate what the link represents?